[WPE][GTK] xdgmime: Define LITTLE_ENDIAN

sideeffect42 opened the pull request:
#### 41089e03042e9aa72099894ba10c19f323e9205a
<pre>
[WPE][GTK] xdgmime: Define LITTLE_ENDIAN
<a href="https://bugs.webkit.org/show_bug.cgi?id=276778">https://bugs.webkit.org/show_bug.cgi?id=276778</a>

Reviewed by Michael Catanzaro.

* Source/ThirdParty/xdgmime/CMakeLists.txt: define LITTLE_ENDIAN
* Source/ThirdParty/xdgmime/src/xdgmimemagic.c: backport upstream commit 92f6a09f

Canonical link: <a href="https://commits.webkit.org/281138@main">https://commits.webkit.org/281138@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8d39f461058913c066f685cc77b3a10823035ca0

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/58853 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/38181 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/11344 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/62485 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/9297 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/45818 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/9498 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/47603 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/6625 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/60884 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/35724 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/50897 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/28460 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/32457 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/8184 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/8301 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/54412 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/8464 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/64186 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/2766 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/8445 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/54927 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/2775 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/50920 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/55022 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13021 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/2320 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/34011 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/35095 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/36180 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/34841 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro reviewed (approved):
Good.

It's probably even easier to change xdgmime to test the `__BYTE_ORDER__` built-in macro, but this is probably more portable, and doesn't require changing any upstream files.

mcatanzaro commented on Source/ThirdParty/xdgmime/CMakeLists.txt:
```diff
@@ -20,9 +20,26 @@ include_directories("${XDGMIME_INCLUDE_DIRECTORIES}")
+    else ()
+        set(CMAKE_CXX_BYTE_ORDER "LITTLE_ENDIAN")
+    endif ()
+endif ()
```
Actually, wait: our `CMAKE_MINIMUM_REQUIRED` is 3.20, so none of this is needed! Delete please.

sideeffect42 replied:
You're right, @mcatanzaro, I hadn't noticed. I added this code because I saw the checks for version 3.20 in `OptionsGTK.cmake`.

I grepped the CMake files for version checks and removed the conditional code for CMake < 3.20.

I hope it's okay to do it in the same PR.

Commit:
[WPE][GTK] xdgmime: Define LITTLE_ENDIAN
https://bugs.webkit.org/show_bug.cgi?id=276778

Reviewed by Michael Catanzaro.

* Source/ThirdParty/xdgmime/CMakeLists.txt: define LITTLE_ENDIAN
* Source/ThirdParty/xdgmime/src/xdgmimemagic.c: backport upstream commit 92f6a09f

Canonical link: https://commits.webkit.org/281138@main
